codersdk/audit.go

```diff
@@ -101,9 +107,22 @@ type AuditLogCountResponse struct {
 	Count int64 `json:"count"`
 }
 
+type CreateTestAuditLogRequest struct {
+	Action       AuditAction  `json:"action,omitempty"`
+	ResourceType ResourceType `json:"resource_type,omitempty"`
+}
+
 // AuditLogs retrieves audit logs from the given page.
-func (c *Client) AuditLogs(ctx context.Context, page Pagination) (AuditLogResponse, error) {
-	res, err := c.Request(ctx, http.MethodGet, "/api/v2/audit", nil, page.asRequestOption())
+func (c *Client) AuditLogs(ctx context.Context, req AuditLogsRequest) (AuditLogResponse, error) {
+	res, err := c.Request(ctx, http.MethodGet, "/api/v2/audit", nil, req.Pagination.asRequestOption(), func(r *http.Request) {
+		q := r.URL.Query()
+		var params []string
+		if req.SearchQuery != "" {
+			params = append(params, req.SearchQuery)
+		}
+		q.Set("q", strings.Join(params, " "))
```

Comment:
I'm slightly confused with `params`. It seems like it can only ever have 0 or 1 elements. Why do we need to `strings.Join` it?

Reply:
Good catch, I copied and paste from the workspace logic.

Reply:
I mean, it could have `resource_type:workspace action:create` right?

Reply:
Ahh, I think I got it. So it should be only:
```
q := r.URL.Query()
r.URL.RawQuery = q.Encode()
```

Reply:
Is this where we're defining the name of the param? Do we have control over calling it `q` or something else?